enhancements/accelerators/out-of-tree-drivers.md

```diff
@@ -0,0 +1,418 @@
+In OpenShift v4.x this changed with the introduction of RHCOS. There are currently two different documented ways to enable out-of-tree drivers. One way is using a SRO based operator such as the nvidia GPU operator and the other is using kmods-via-containers. 
+We want to come up with a unified solution that works for our customers across RHCOS and RHEL. The solution should also help customers that are currently on RHEL7 to consider moving to RHCOS which is fully managed and easier to support in OpenShift.
+ 
+#### Day-1 DriverContainer Management  [OpenShift | BareMetal]
+ 
+TODO: LONGER TERM
+ 
+#### Day-2 DriverContainer Management OpenShift
+ 
+More and more customers/partners want to enable hardware and or software on OpenShift that need kernel drivers, which are currently not (and may never be) upstream   and out-of-tree. 
+We want to provide a unified way to support out-of-tree kernel drivers (multiple) on OpenShift for day-1 (long term) and day-2 (short term). It has to work for classical (RHEL7, 8, Fedora) and container based operating systems (RHCOS, FCOS) in the same way. 
+Many customers/partners are using dkms/akmods as the solution to build, rebuild modules on kernel changes. Adopting dkms is not a solution; we need to  create an own acceptable solution for RHEL and Fedora. 
+ 
+#### Fill the gap of providing drivers that are not yet upstream or never be
+
+For the DriverContainer we need to cover several stages of driver packaging. 
+
+* **github repository or archive** The driver is only available as  source code and is not packaged in any way, this is where KVC can help
+* **kmod-{vendor}-src.rpm** The next step is a source RPM package that can be recompiled with rpmbuild. This is also the base for akmods, dkms
+* **precompiled-{vendor}-{kernelversion}.rpm** Precompiled RPMS this is the wish thinking to have in the future. DriverContainers could be easily built just by using RPMs. 
+
+Some drivers will *never* be upstreamed and can be in any of the states described above.  The proposed solutions needs to handle drivers in any state and build by "any" tool. 
+
+The compilation of kernel modules was always anticipated to be the fallback solution when dealing with kernel modules. Some kernel modules will always be out-of-tree and are not going to be included in the near future. Some kernel modules are out-of-tree but we are working with the vendors on upstreaming them  to the mainline kernel. 
+This work is being done by a virtual-team consisting of an  OpenShift-accelerators  and RHEL-accelerator team responsible for enabling hardware and software on RHEL and OpenShift. 
+
+
+### Goals
+
+
+* A unified way to deploy out-of-tree drivers on OpenShift 4.x on all supported Red Hat Operating Systems
+* The solution should avoid rebuilds on every node and allow for distribution of drivers on a cluster using the cluster registry
+* A solution for day-1 and day-2 kernel modules
```

Comment:
The meaning of Day-1 and Day-2 kernel modules needs to be fleshed out a lot more.